### PR TITLE
Streamline and modernize naming

### PR DESCRIPTION
**Pull request by wycats**
This PR pulls out the naming and documentation changes in #1649 so they can be landed separately.

I just finished resolving all of the merge conflicts, so I expect tests, types and lints to fail at the moment. 

It may still be appropriate to break this PR up further. Once tests are passing, we can evaluate that question.

[NOTE] This PR builds on #1664, and will be a bit smaller once that lands.

### CODE REVIEW

File: mise.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[tools]
```
Comment (NullVoxPopuli) [inline mise.toml:1]: isn't the point of these tools (mise, proto, etc) to not need tool-specific configs in the repo?

the packageManager field in the root package.json should be sufficient, yeah?
and then an engines entry?

Comment (wycats) [reply mise.toml:1]: > isn't the point of these tools (mise, proto, etc) to not need tool-specific configs in the repo?

Whether or not this is true, it's an accident that this got pushed.

File: packages/@glimmer/interfaces/lib/dom/attributes.d.ts [hunk 2]
```diff
@@ -12,23 +12,30 @@ import type {
...
-export interface SimpleLiveBlock extends LiveBlock {
-  parentElement(): SimpleElement;
-  firstNode(): SimpleNode;
-  lastNode(): SimpleNode;
-}
-
-export type RemoteLiveBlock = SimpleLiveBlock;
-
-export interface UpdatableBlock extends SimpleLiveBlock {
+/**
```
Comment (NullVoxPopuli) [inline packages/@glimmer/interfaces/lib/dom/attributes.d.ts:23]: I like these much better than their previous names

File: packages/@glimmer/interfaces/lib/program.d.ts [hunk 3]
```diff
@@ -44,36 +41,47 @@ export interface CompileTimeHeap extends OpcodeHeap {
...
+   * The compiled program itself: the constants and heap
+   */
+  readonly program: Program;
+  /**
+   * The offsets to stdlib functions
+   */
   readonly stdlib: STDLib;
-
+  /**
+   * A framework-specified resolver for resolving free variables in classic templates.
```
Comment (NullVoxPopuli) [inline packages/@glimmer/interfaces/lib/program.d.ts:65]: I super appreciate all these inline comments!

**Review by NullVoxPopuli — APPROVED** [review]
lgtm!

**Pull request merged**